### ISSUE TITLE
正常图片和异常图片得到的分类概率分数都一样 

### ISSUE THREAD

**Issue by cclamd** [ByChelsea/VAND-APRIL-GAN#10]
你好， 这边做测试用一张正常图片和一张异常图片做测试 代码如下

`image = preprocess(Image.open("/content/000.png")).unsqueeze(0).to(device)
obj_list=["screw"]
with torch.cuda.amp.autocast(), torch.no_grad():
    text_prompts = encode_text_with_prompt_ensemble(model, obj_list, tokenizer, device) 
    image_features= model.encode_image(image)
    image_features /= image_features.norm(dim=-1, keepdim=True)

    text_features = []
    text_features.append(text_prompts["screw"])
    text_features = torch.stack(text_features, dim=0)
    # sample
    text_probs = (100.0 * image_features @ text_features[0]).softmax(dim=-1)`

得到的text_probs 都为这样tensor([[0.7982, 0.2018]], device='cuda:0') ，第一个值都是大于第二值，请问这样是否表明分类不正确

**Comment by ByChelsea** [comment]
Hi! The results like these do not necessarily imply incorrect classification. In fact, we use the scores **associated with text describing anomalies** as anomaly scores and determine the presence of anomalies based on the **relative magnitude** of these anomaly scores.

It's pretty challenging to perform classification solely based on the similarity between image and text features. We believe this difficulty may stem from CLIP's limited sensitivity in discerning between normal and abnormal cases. Designing more appropriate prompts may alleviate this issue, and prompt learning could also be considered.

**Comment by cclamd** [comment]
I understand, thank you.
